shielddb-gson-backend/src/test/java/nf/fr/k49/shielddb/gson/ShieldDBGsonTest.java

```diff
@@ -0,0 +1,44 @@
+package nf.fr.k49.shielddb.gson;
+
+import com.google.gson.Gson;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class ShieldDBGsonTest {
+
+    @Test
+    public void emptyArrayListWillBeReturnedOnProvidedNullJSON() {
+        ShieldDBGson<String> shieldDBGson = new ShieldDBGson<>(new Gson(), String.class);
+
+        List<String> list = shieldDBGson.toList(null);
+
+        assertThat(list.size(), is(0));
+    }
+
+    @Test
+    public void emptyArrayListWillBeReturnedOnProvidedEmptyJSON() {
+        ShieldDBGson<String> shieldDBGson = new ShieldDBGson<>(new Gson(), String.class);
+
+        List<String> list = shieldDBGson.toList("");
+
+        assertThat(list.size(), is(0));
+    }
+
+    @Test
+    public void correctArrayListWillBeReturnedOnProvidedJSONArray() {
+        String json = "[ \"String1\", \"String2\", \"String3\" ]";
+
+        ShieldDBGson<String> shieldDBGson = new ShieldDBGson<>(new Gson(), GsonTypeUtils.getType());
+
+        List<String> list = shieldDBGson.toList(json);
+
+        assertThat(list.size(), is(3));
+        assertThat(list, containsInAnyOrder("String1", "String2", "String3"));
+    }
+
```

Comment:
you did not test the toJson() method. Can you add tests on this method?

Reply:
First point: today it contains only one line of code, but in the future it may not.
Second point: best unit tests are black boxed, so you should write tests as you did not read the code.
Third point: test are for me a living documentation, so do not testing a method is like they do nothing, so testing ShieldDBGson involve you to test every public method, this way anywhere we will use this method, if we change toJson(), tests will tell us what was expected.